source/draw/DrawState.ooc

```diff
@@ -0,0 +1,79 @@
+//
+// Copyright (c) 2011-2014 Simon Mika <simon@mika.se>
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+use geometry
+import Pen, Image, Map
+
+// Example
+// DrawState new(myImage) setMap(myShader) draw()
+
+DrawState: cover {
+	target: Image = null
+	inputImage: Image = null
+	map: Map = null
+	opacity := 1.0f
+	_transformNormalized := FloatTransform3D identity
+	viewport := IntBox2D new(0, 0, 0, 0)
+	init: func@ ~default
+	init: func@ ~target (=target)
+	setTarget: func (target: Image) -> This {
+		result := this
+		result target = target
+		result
+	}
```

Comment:
I mean the `result := this` part is not neccessary, `this` is already passed as a copy because the call is made on a cover (and the function signature does not have a reference indicator `@`).